src/App.jsx

```diff
@@ -14,33 +15,97 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [activeSort, setActiveSort] = useState(null);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort(null);
+  };
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const sortAlphabetically = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort((a, b) =>
+        a.localeCompare(b),
+      );
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      if (isReversed) {
+        return sortedGoods.reverse();
+      }
+
+      return sortedGoods;
+    });
+    setActiveSort('alphabetical');
+  };
+
+  const sortByLength = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort(
+        (a, b) => a.length - b.length,
+      );
+
+      if (isReversed) {
+        return sortedGoods.reverse();
+      }
+
+      return sortedGoods;
+    });
+    setActiveSort('length');
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevReversed => !prevReversed);
```

Comment:
The `reverseGoods` function reverses the current state of `goods` and toggles `isReversed`. However, when combined with sorting, the `isReversed` state might not accurately reflect the current order of `goods`. Consider ensuring that the `isReversed` state is correctly managed in conjunction with sorting functions to maintain consistency.